python/pyspark/pandas/spark/accessors.py

```diff
@@ -64,7 +68,7 @@ def column(self) -> Column:
         """
         return self._data._internal.spark_column_for(self._data._column_label)
 
-    def transform(self, func: Callable[[Column], Column]) -> IndexOpsLike:
+    def transform(self, func: Callable[[Column], Union[Column, ConnectColumn]]) -> IndexOpsLike:
```

Comment:
`Callable[[GenericColumn], GenericColumn]`?

Reply:
ditto, let me address in subsequent PR since it brings another hundreds of mypy failures? I plan to create several subsequent tickets after completing the initial PR, so I'd want to focus on setting the test base first here in this PR and encourage community involvement for the remaining errands.